File: 001. Two Sum/1. Two Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var idx: [Int] = [1, 3]
        var i = 0, j = 1
        while i < nums.count - 1 {
            j = i + 1
            while j < nums.count {
                if nums[i] + nums[j] == target {
                    idx[0] = i
                    idx[1] = j
                    return idx
                }
                j += 1
            }
            i += 1
        }
        return idx
    }
}

File: 001. Two Sum/1. Two Sum.cpp
class Solution { 
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0, n = nums.size();
        unordered_map<int, int> idx; 
        while (i < n && idx.find(target - nums[i]) == idx.end()) { // find the other number and not same with the current number
            idx[nums[i++]] = i;
        }
    return {idx[target - nums[i]] - 1, i};
    }
};
